main.rb

```diff
@@ -0,0 +1,160 @@
+# enumerable methods
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    my_arr = *self
+    i = 0
+    while i < my_arr.length
+      yield my_arr[i]
+      i += 1
+    end
+    my_arr
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    my_arr = *self
+    i = 0
+    while i < my_arr.length
+      yield my_arr[i], i
+      i += 1
+    end
+    my_arr
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    result = []
+    my_each { |x| result << x if yield(x) }
+    result
+  end
+
+  # rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+
+  def my_all?(pattern = nil)
+    my_each do |x|
+      if block_given?
+        return false unless yield x
+      elsif pattern.class == Regexp
+        return false unless pattern =~ x
+      elsif pattern.class == Class
+        return false unless x.class == pattern
+      elsif !pattern.nil?
+        return false unless x == pattern
+      else
+        return false unless x
+      end
+    end
+    true
+  end
+
+  def my_any?(pattern = nil)
+    return false if (self - [nil, false]) == []
+
+    my_each do |x|
+      if block_given?
+        return true if yield x
+      elsif pattern.class == Regexp
+        return true if pattern =~ x
+      elsif pattern.class == Class
+        return true if x.class == pattern
+      elsif !pattern.nil?
+        return true if x == pattern
+      else
+        return true unless x
+      end
+    end
+    return true if pattern.nil? && !block_given?
+
+    false
+  end

```

Comment:
- [ ] my_any? returns false if the block ever returns a value false or nil. This method is working OK for an array, but it throws an error when calling it from a range.

Example:

`(1..10).my_any? { | value | value > 10 }` should return false
`(1..11).my_any? { | value | value > 10 }` should return true

Print from the original method:
![Screenshot from 2020-06-05 09-44-27](https://user-images.githubusercontent.com/30977845/83877606-3002e480-a711-11ea-98f8-cc384561c6c0.png)


- [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class.

`[1, 2.14, 0.5, 0].my_any?(Integer)` should return true
`[2.14, 0.5].my_any?(Integer)` should return false
`[1, 2.14, 0.5, 'hello'].my_any?(String)` should return true